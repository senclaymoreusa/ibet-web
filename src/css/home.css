.games {
    font-size: 30px;
    margin-left: 40%;
}

.poker {
    font-size: 30px;
    margin-left: 40%;
}

.casino {
    font-size: 30px;
    margin-left: 40%;
}

.each-game {
    margin: auto;
    text-align: center;
}


