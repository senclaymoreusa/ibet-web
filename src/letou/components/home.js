/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import Footer from './footer';
import TopNavbar from './top_navbar';
import { injectIntl } from 'react-intl';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  authCheckState,
  handle_referid,
  hide_landing_page
} from '../../actions';
import { withStyles } from '@material-ui/core/styles';
import { Slide } from 'react-slideshow-image';
import { images } from '../../util_config';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import axios from 'axios';

document.body.style = 'background: #f1f1f1;';

const API_URL = process.env.REACT_APP_DEVELOP_API_URL;

const styles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
    backgroundColor: theme.palette.background.paper
  },
  bannerContainer: {
    height: 504
  },
  banner: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundSize: 'cover',
    height: 500
  },
  container: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  contentDesktop: {
    maxWidth: 1400,
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
  },
  leftColumn: {
    width: 260
  },
  rightColumn: {
    marginLeft: 5
  },
  left: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundSize: 'cover',
    height: 260
  },
  grow: {
    flexGrow: 1
  },
  subTitle: {
    fontSize: 18,
    color: '#212121',
    fontWeight: 500,
    fontStyle: 'normal',
    lineHeight: 1.57
  },
  desc: {
    fontSize: 14,
    color: '#212121',
    fontWeight: 500,
    fontStyle: 'normal',
  },
  text: {
    fontSize: 14,
    color: '#212121',
    fontWeight: 'normal',
    fontStyle: 'normal',
    lineHeight: 1.57
  },
  row: {
    border: '1px solid #f0f0f0',
    marginBottom: 10,
    backgroundColor: 'white',
    '&:hover': {
      backgroundColor: '#f9f9f9',
    },
  },
});

const ColorButton = withStyles(theme => ({
  root: {
    color: 'white',
    backgroundColor: '#ff9e00',
    '&:hover': {
      backgroundColor: '#ff9e00',
    },
  },
}))(Button);

export class Home extends Component {

  constructor(props) {
    super(props);

    this.searchDiv = React.createRef();

    this.state = {
      width: 800
    };
  }

  handleResize = () => {
    this.setState({ height: window.innerHeight, width: window.innerWidth })
  };

  componentWillReceiveProps(props) {
    window.addEventListener("resize", this.handleResize);
  }

  componentDidMount() {
    this.setState({ width: window.innerWidth });

    window.addEventListener("resize", this.handleResize);

    

    var bbData = window.IGLOO.getBlackbox();
    
    if (bbData.finished) {
      // clearTimeout(timeoutId);
      var blackBoxString = bbData.blackbox;
      axios
          .get(
              API_URL + 'users/api/login-device-info?bb=' + blackBoxString
          ).then(
            res => {
              //console.log(res.data);
              const ip = res.data.details.realIp.address;
              const countryCode = res.data.details.realIp.ipLocation.countryCode;
              
              localStorage.setItem('ip', ip);
              localStorage.setItem('countryCode', countryCode);

              // var user_ip = localStorage.getItem('ip');
              // console.log(user_ip);

            }
          );
    }
  }

  render() {
    const { classes } = this.props;

    const bannerImages = [
      images.src + 'letou/banner1.jpg',
      images.src + 'letou/banner2.jpg',
      images.src + 'letou/banner3.jpg',
      images.src + 'letou/banner4.jpg',
      images.src + 'letou/banner5.jpg',
    ];

    const bannerProperties = {
      duration: 2000,
      transitionDuration: 500,
      infinite: true,
      indicators: true,
      arrows: true
    }

    const leftProperties = {
      duration: 3000,
      transitionDuration: 50,
      infinite: true,
      indicators: false,
      arrows: false,
      autoplay: true
    }

    return (
      <div className={classes.root}>
        <TopNavbar />
        <div className={classes.bannerContainer}>
          <Slide {...bannerProperties}>
            {bannerImages.map(item => (
              <div key={item} className={classes.banner} style={{ 'backgroundImage': `url(${item})` }} />
            ))}
          </Slide>
        </div>
        <div className={classes.container}>
          <div className={classes.contentDesktop}>
            <div className={classes.leftColumn}>
              <Grid container>
                <Grid item xs={12} style={{ paddingBottom: 20, borderBottom: '3px solid #ff9e00' }}>
                  <Slide {...leftProperties}>
                    <div className={classes.left} style={{ 'backgroundImage': `url(${images.src + 'letou/left1.jpg'})` }} />
                    <div className={classes.left} style={{ 'backgroundImage': `url(${images.src + 'letou/left2.jpg'})` }} />
                  </Slide>
                </Grid>
                <Grid item xs={12} style={{ paddingBottom: 10, paddingTop: 10 }}>
                  <span className={classes.subTitle}>Coming soon</span>
                </Grid>
                <Grid item xs={12}>
                  <img
                    src={
                      images.src +
                      'letou/left3.jpg'
                    }
                    alt=""
                  />
                </Grid>
                <Grid item xs={12} style={{ backgroundColor: '#fafafa', padding: 10 }}>
                  <span className={classes.desc}>Boston Celtics v Detroit Pistons</span>
                </Grid>
                <Grid item xs={12} style={{ border: '1px solid #f3f3f3', padding: 10 }}>
                  <span className={classes.desc}>Boston Celtics</span>
                </Grid>
                <Grid item xs={12} style={{ border: '1px solid #f3f3f3', padding: 10 }}>
                  <span className={classes.desc}>Boston Celtics</span>
                </Grid>
                <Grid item xs={12} style={{ borderBottom: '3px solid #ff9e00', height: 20 }}>
                </Grid>
                <Grid item xs={12} style={{ paddingBottom: 10, paddingTop: 10 }}>
                  <span className={classes.subTitle}>Digital color</span>
                </Grid>
                <Grid item xs={6} style={{ borderTop: '1px solid #3eace9', borderLeft: '1px solid #3eace9', borderBottom: '1px solid #3eace9', height: 32 }}>
                  <img
                    src={
                      images.src +
                      'letou/degitalpic1.jpg'
                    }
                    height={42}
                    width={130}
                    alt=""
                  />
                </Grid>
                <Grid item xs={6} style={{ borderTop: '1px solid #3eace9', borderRight: '1px solid #3eace9', borderBottom: '1px solid #3eace9', height: 42, color: '#3eace9', paddingTop: 8, paddingLeft: 10 }}>
                  <span >Happy color</span>
                </Grid>
                <Grid item xs={6} style={{ borderLeft: '1px solid #3eace9', borderBottom: '1px solid #3eace9', height: 42 }}>
                  <img
                    src={
                      images.src +
                      'letou/degitalpic2.jpg'
                    }
                    height={42}
                    width={130}
                    alt=""
                  />
                </Grid>
                <Grid item xs={6} style={{ borderRight: '1px solid #3eace9', borderBottom: '1px solid #3eace9', height: 42, color: '#3eace9', paddingTop: 8, paddingLeft: 10 }}>
                  <span >Lotto</span>
                </Grid>
                <Grid item xs={12} style={{ borderBottom: '3px solid #ff9e00', height: 20 }}>
                </Grid>
                <Grid item xs={12} style={{ paddingBottom: 10, paddingTop: 10 }}>
                  <span className={classes.subTitle}>Live dealer</span>
                </Grid>
                <Grid item xs={6} style={{ borderTop: '1px solid #3eace9', borderLeft: '1px solid #3eace9', borderBottom: '1px solid #3eace9', height: 42 }}>
                  <img
                    src={
                      images.src +
                      'letou/pgbanner1.jpg'
                    }
                    height={42}
                    width={130}
                    alt=""
                  />
                </Grid>
                <Grid item xs={6} style={{ borderTop: '1px solid #3eace9', borderRight: '1px solid #3eace9', borderBottom: '1px solid #3eace9', height: 42, color: '#3eace9', paddingTop: 8, paddingLeft: 10 }}>
                  <span >Baccarat</span>
                </Grid>
                <Grid item xs={12} style={{ borderBottom: '3px solid #ff9e00', height: 20 }}>
                </Grid>
                <Grid item xs={12} style={{ paddingBottom: 10, paddingTop: 10 }}>
                  <span className={classes.subTitle}>Latest Announcement</span>
                </Grid>
                <Grid item xs={12} style={{ paddingBottom: 10, paddingTop: 10, borderTop: '3px solid #f3f3f3' }}>
                  <span className={classes.desc}>¥ Central People’s Bank of China...</span>
                </Grid>
              </Grid>
            </div>
            <div className={classes.rightColumn}>
              <Grid container>
                <Grid item xs={12} style={{ padding: 20, display: 'flex', flexDirection: 'row', backgroundColor: 'white' }}>
                  <span className={classes.subTitle}>Letou Featured Promotions</span>
                  <span className={classes.desc} style={{ marginLeft: 20, marginTop: 6 }}>Various offers to support you</span>
                </Grid>
                <Grid item xs={4} style={{ padding: 20, display: 'flex', flexDirection: 'column', }}>
                  <img
                    src={
                      images.src +
                      'letou/c1.jpg'
                    }
                    alt=""
                    height={147}
                  />
                  <span className={classes.text} style={{ marginTop: 20 }}>Le Tou crazy football! All the leagues in all countries have made every week's winnings and fully upgraded!</span>
                </Grid>
                <Grid item xs={4} style={{ padding: 20, display: 'flex', flexDirection: 'column' }}>
                  <img
                    src={
                      images.src +
                      'letou/c2.jpg'
                    }
                    height={147}
                    alt=""
                  />
                  <span className={classes.text} style={{ marginTop: 20 }}>Le Tou crazy football! All the leagues in all countries have made every week's winnings and fully upgraded!</span>
                </Grid>
                <Grid item xs={4} style={{ padding: 20, display: 'flex', flexDirection: 'column', }}>
                  <img
                    src={
                      images.src +
                      'letou/c3.jpg'
                    }
                    alt=""
                    height={147}
                  />
                  <span className={classes.text} style={{ marginTop: 20 }}>Le Tou crazy football! All the leagues in all countries have made every week's winnings and fully upgraded!</span>
                </Grid>
                <Grid item xs={12} style={{ paddingLeft: 20, marginBottom:20, display: 'flex', flexDirection: 'row', borderBottom: '2px solid #f3f3f3' }}>
                  <span className={classes.subTitle}>Top Games</span>
                  <div className={classes.grow} />
                  <ColorButton style={{marginBottom:10}}>More</ColorButton>
                </Grid>
                <Grid item xs={12} className={classes.row} style={{ padding: 20, display: 'flex', flexDirection: 'row', cursor: 'pointer' }}>
                  <img
                    src={
                      images.src +
                      'letou/ct1.jpg'
                    }
                    style={{ minWidth: 221 }}
                    height={168}
                    width={221}
                    alt=""
                  />
                  <div style={{ padding: 20, display: 'flex', flexDirection: 'column' }}>
                    <span className={classes.subTitle}>Captain's Treasure</span>
                    <span className={classes.text}>Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times! Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times! Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times!</span>
                  </div>
                </Grid>
                <Grid item xs={12} className={classes.row} style={{ padding: 20, display: 'flex', flexDirection: 'row', cursor: 'pointer' }}>
                  <img
                    src={
                      images.src +
                      'letou/ct2.jpg'
                    }
                    style={{ minWidth: 221 }}
                    height={168}
                    width={221}
                    alt=""
                  />
                  <div style={{ padding: 20, display: 'flex', flexDirection: 'column' }}>
                    <span className={classes.subTitle}>Line 5-Fruit Festival</span>
                    <span className={classes.text}>Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times! Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times! Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times!</span>
                  </div>
                </Grid>
                <Grid item xs={12} className={classes.row} style={{ padding: 20, display: 'flex', flexDirection: 'row', cursor: 'pointer', '&:hover': { backgroundColor: 'orange' } }}>
                  <img
                    src={
                      images.src +
                      'letou/ct3.jpg'
                    }
                    style={{ minWidth: 221 }}
                    height={168}
                    width={221}
                    alt=""
                  />
                  <div style={{ padding: 20, display: 'flex', flexDirection: 'column' }}>
                    <span className={classes.subTitle}>More than every year</span>
                    <span className={classes.text}>Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times! Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times! Take a sea adventure with Captain Pirate! The game line can be up to 50 and can be up to 25 lines. The highest multiple is 10,000 times!</span>
                  </div>
                </Grid>
              </Grid>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    lang: state.language.lang
  };
};

export default withStyles(styles)(
  injectIntl(
    withRouter(
      connect(mapStateToProps, {
        authCheckState,
        handle_referid,
        hide_landing_page
      })(Home)
    )
  )
);
