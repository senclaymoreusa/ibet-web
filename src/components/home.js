import React, { Component } from 'react';
import Footer from "./footer";
import TopNavbar from "./top_navbar";
import { connect } from 'react-redux';
import { authCheckState } from '../actions';
import { FormattedMessage } from 'react-intl';
import Marquee from "react-smooth-marquee";
import axios from 'axios';
import { config } from '../util_config';
import moment from 'moment';
import { NavLink } from 'react-router-dom';
import '../css/home.css';


const API_URL = process.env.REACT_APP_REST_API;

var height = window.innerHeight

export class Home extends Component {

  state = {
    notices: [],
    sports: [],
    casino: [],
    poker: [],
    ready: false
  }

  async componentDidMount() {

    this.props.authCheckState()
    axios.get(API_URL + 'operation/api/notice-message', config)
      .then(res => {
        //   console.log(res);
        this.setState({ notices: res.data });
      })

    var URL = API_URL + 'users/api/games/?term=Sports';

    await axios.get(URL, config)
      .then(res => {
        this.setState({ sports: res.data.slice(0, 3) });
      })

    URL = API_URL + 'users/api/games/?term=Casino';

    await axios.get(URL, config)
      .then(res => {
        this.setState({ casino: res.data.slice(0, 3) });
      })

    URL = API_URL + 'users/api/games/?term=Poker';

    await axios.get(URL, config)
      .then(res => {
        this.setState({ poker: res.data.slice(0, 3) });
      })

    this.setState({ ready: true })

  }

  render() {

    let notices = this.state.notices;

    let noticeStr = '';
    notices.forEach(notice => {
      let startTime = moment(notice.start_time);
      startTime = startTime.format('MM/DD/YYYY h:mm a');
      let endTime = moment(notice.end_time);
      endTime = endTime.format('MM/DD/YYYY h:mm a');
      let i18nMessage = notice.message;
      if (this.props.lang === 'zh') {
        i18nMessage = notice.message_zh;
      } else if (this.props.lang === 'fr') {
        i18nMessage = notice.message_fr;
      } else {
        i18nMessage = notice.message;
      }
      let message = startTime + " ~ " + endTime + " " + i18nMessage;
      noticeStr += message;
      for (let i = 0; i < 20; i++) {
        noticeStr += "\u00A0";
      }
    });

    return (
      <div >
        <TopNavbar style={{zIndex: '100'}} />
        
        {noticeStr && <div style={{ overflowX: 'hidden', zIndex: 0 }}><Marquee >{noticeStr}</Marquee></div>}

        {
          this.state.ready &&
          <div className='games' style={{ marginTop: height * 0.1 }}>
            <FormattedMessage id="home.sports" defaultMessage='Most Popular Sports' />
          </div>
        }

        <div className="rows" >
          {
            this.state.ready && this.state.sports.map(item => {
              return (
                  <div key={item.name} className='each-game' >
                    <NavLink to = {`/game_detail/${item.pk}`} style={{ textDecoration: 'none' }} onClick={()=>{
                        }}> {item.name} </NavLink>
                    <br/>
                    <img src={item.image} height = "100" width="100" alt = 'Not available'/>
                  </div>
                )
            })
          }
        </div>


        {
          this.state.ready &&
          <div className='poker' style={{ marginTop: height * 0.1 }}>
            <FormattedMessage id="home.poker" defaultMessage='Most Popluar Poker' />
          </div>
        }

        <div className="rows" >
          {
            this.state.ready && this.state.poker.map(item => {
              return (
                  <div key={item.name} className='each-game'>
                    <NavLink to = {`/game_detail/${item.pk}`} style={{ textDecoration: 'none' }} onClick={()=>{
                        }}> {item.name} </NavLink>
                    <br/>
                    <img src={item.image} height = "100" width="100" alt = 'Not available'/>
                  </div>
                )
            })
          }
        </div>

        {
          this.state.ready &&
          <div className='casino' style={{ marginTop: height * 0.1 }}>
            <FormattedMessage id="home.casino" defaultMessage='Most Popluar Casino' />
          </div>
        }

        <div className="rows" >
          {
            this.state.ready && this.state.casino.map(item => {
              return (
                  <div key={item.name} className='each-game'>
                    <NavLink to = {`/game_detail/${item.pk}`} style={{ textDecoration: 'none' }} onClick={()=>{
                        }}> {item.name} </NavLink>
                    <br/>
                    <img src={item.image} height = "100" width="100" alt = 'Not available'/>
                  </div>
                )
            })
          }
        </div>
          <Footer/>
      </div>

    );
  }
}

const mapStateToProps = (state) => {
  return {
    lang: state.language.lang
  }
}

export default connect(mapStateToProps, { authCheckState })(Home);