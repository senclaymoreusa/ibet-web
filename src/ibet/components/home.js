import React, { Component } from 'react';
import Footer from "./footer";
import TopNavbar from "./top_navbar";
import ChatTool from "./chat_tool";
import { connect } from 'react-redux';
import { authCheckState, handle_referid, hide_landing_page, sendingLog } from '../../actions';
import { FormattedMessage } from 'react-intl';
import { config } from '../../util_config';

import axios from 'axios';
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import ExpandMore from '@material-ui/icons/ExpandMore';

import '../css/home.css';

const API_URL = process.env.REACT_APP_DEVELOP_API_URL


console.log("Line 15, process env URL = " + API_URL);

document.body.style = 'background: #f1f1f1;';


const styles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
    backgroundColor: theme.palette.background.paper,

  },
  fab: {
    width: '240px',
    marginTop: '48px',
    backgroundColor: '#ffffff;',
    fontSize: '18px'
  },
  extendedIcon: {
    marginRight: theme.spacing(),
  },
  grow: {
    flexGrow: 1,
  },
});

export class Home extends Component {

  state = {
    notices: [],
    sports: [],
    casino: [],
    poker: [],
    ready: false,
    expand: false,

    all_sports: []
  }

  async componentDidMount() {
    const { referid } = this.props.match.params;
    if (referid) {
      this.props.handle_referid(referid);
    }

    this.props.authCheckState();
    var URL = API_URL + 'users/api/games/?term=game';

    await axios.get(URL, config)
      .then(res => {

        this.setState({ sports: res.data.slice(0, 8) });
        this.setState({ all_sports: res.data })
      })

    URL = API_URL + 'users/api/games/?term=Casino';

    await axios.get(URL, config)
      .then(res => {
        this.setState({ casino: res.data.slice(0, 3) });
      })

    URL = API_URL + 'users/api/games/?term=Poker';

    await axios.get(URL, config)
      .then(res => {
        this.setState({ poker: res.data.slice(0, 3) });
      })

    this.setState({ ready: true })

  }

  handle_expand() {
    this.setState({ sports: this.state.all_sports })
    this.setState({ expand: true })
  }

  game_url(event, item){
    event.preventDefault();
    console.log(item.name)
    if (['SA', 'Lotto', 'Keno', 'PK10', 'K3', 'SSC', 'GB'].includes(item.name)){
      console.log('12345')
      var token = localStorage.getItem('token')
      if (token){
          config.headers["Authorization"] = `Token ${token}`;
          var URL = API_URL + 'users/api/generategameurl/'
          axios.get(URL, config)
          .then(res => {
             console.log(res)
              var Game_URL = res.data.game_url
              console.log(Game_URL)
              window.open(Game_URL)
          })
      }else{
        alert('You have to login to play this game')
      }

    }else{
      window.open(item.game_url)
    }
  }

  render() {

    const { classes } = this.props;

    // let notices = this.state.notices;

    // let noticeStr = '';
    // notices.forEach(notice => {
    //   let startTime = moment(notice.start_time);
    //   startTime = startTime.format('MM/DD/YYYY h:mm a');
    //   let endTime = moment(notice.end_time);
    //   endTime = endTime.format('MM/DD/YYYY h:mm a');
    //   let i18nMessage = notice.message;
    //   if (this.props.lang === 'zh') {
    //     i18nMessage = notice.message_zh;
    //   } else if (this.props.lang === 'fr') {
    //     i18nMessage = notice.message_fr;
    //   } else {
    //     i18nMessage = notice.message;
    //   }
    //   let message = startTime + " ~ " + endTime + " " + i18nMessage;
    //   noticeStr += message;
    //   for (let i = 0; i < 20; i++) {
    //     noticeStr += "\u00A0";
    //   }
    // });

    var recent_games = JSON.parse(localStorage.getItem("recent-games"));

    return (
      <div className={classes.root}>
        {/* {
          this.props.show_landing_page && <Paper >
            <div style={{ position: 'absolute', zIndex: 1000000000000 }} onClick={() => {
              this.props.hide_landing_page()
            }}>
              <img src='https://ibet-web.s3-us-west-1.amazonaws.com/Games/landing-page.jpg' height="100%" width="100%" alt='Not available' />
            </div>
          </Paper>
        } */}
        <TopNavbar />
        <div className={classes.grow}>

          {
            this.state.ready &&
            <div className='top-title'>
              <FormattedMessage id="home.new" defaultMessage='New Games' />
            </div>
          }

          <div className="cont">

            {
              this.state.ready && this.state.sports.map(item => {
                return (
                  <div key={item.pk} className='each-game' onClick={() => {
                    var array = JSON.parse(localStorage.getItem("recent-games"));
                    if (!array) {
                      array = []
                      array.push(item)
                    } else {
                      var check = true;
                      array.forEach(thing => {
                        if (thing.name === item.name) {
                          check = false
                        }
                      })
                      if (check) {
                        array.push(item)
                      }

                    }
                    if (array.length > 4) {
                      array.shift()
                    }
                    localStorage.setItem("recent-games", JSON.stringify(array));

                  }}>
                    <span onClick={(e) => this.game_url(e, item)}> 

                      {
                        <img src={item.image_url} height="220" width="300" alt='Not available' />
                      }


                      <br />
                      <div className='game-title'>
                        {item.name}
                      </div>

                    </span>
                  </div>
                )
              })
            }

            {/* {
            <div 
              className='each-game' 
              style={{cursor: 'pointer'}}
              onClick={() => {
                this.props.history.push('/game_detail/')
              }}>
              <span >

              {
                <img src={placeholdimage} height="220" width="300" alt='Not available' />
              }


              <br />
              <div className='game-title'>
                AG
              </div>

              </span>
            </div>
          } */}
          </div>


          {
            this.state.ready && !this.state.expand && this.state.all_sports.length > 8 &&
            <div className='expand-icon'>
              <Fab
                onClick={this.handle_expand.bind(this)}
                className={classNames(classes.fab, 'text')}
                variant="extended"
              >
                <FormattedMessage id="home.expand" defaultMessage='View All' />
                {' (' + this.state.all_sports.length + ')'}
                <ExpandMore />
              </Fab>
            </div>
          }


          {
            this.state.ready &&
            <div className='top-title margin-title' >
              <FormattedMessage id="home.recent" defaultMessage='Recently Played' />
            </div>
          }

          <div className="cont">
            {
              this.state.ready && recent_games && recent_games.map(item => {
                return (
                  <div key={item.name} className='each-game' >
                    <span onClick={() => { window.open(item.game_url) }}>

                      {
                        <img src={item.image_url} height="220" width="300" alt='Not available' />
                      }

                      <br />

                      <div className='game-title'>
                        {item.name}
                      </div>

                    </span>
                  </div>
                )
              })
            }
          </div>

          {
            this.state.ready &&
            <div className='top-title margin-title' >
              <FormattedMessage id="home.selected" defaultMessage='Selected for you' />
            </div>
          }

          <div className="cont">
            {
              this.state.ready && this.state.sports.slice(0, 4).map(item => {
                return (
                  <div key={item.name} className='each-game' >
                    <span onClick={() => { window.open(item.game_url) }}>

                      {
                        <img src={item.image_url} height="220" width="300" alt='Not available' />
                      }

                      <br />

                      <div className='game-title'>
                        {item.name}
                      </div>

                    </span>
                  </div>
                )
              })
            }
          </div>



        </div>

        {/* {
          this.state.ready &&
          <div className='poker' style={{ marginTop: height * 0.1 }}>
            <FormattedMessage id="home.poker" defaultMessage='Most Popluar Poker' />
          </div>
        }

        <div className="row" >
          {
            this.state.ready && this.state.poker.map(item => {
              return (
                  <div key={item.name} className='each-game'>
                    <NavLink to = {`/game_detail/${item.pk}`} style={{ textDecoration: 'none' }}> 
                    {item.name} 
                    <br/>
                    <img src={item.image} height = "100" width="100" alt = 'Not available'/>
                    </NavLink>
                  </div>
                )
            })
          }
        </div>

        {
          this.state.ready &&
          <div className='casino' style={{ marginTop: height * 0.1 }}>
            <FormattedMessage id="home.casino" defaultMessage='Most Popluar Casino' />
          </div>
        }

        <div className="row" >
          {
            this.state.ready && this.state.casino.map(item => {
              return (
                  <div key={item.name} className='each-game'>
                    <NavLink to = {`/game_detail/${item.pk}`} style={{ textDecoration: 'none' }}>
                    {item.name} 
                    <br/>
                    <img src={item.image} height = "100" width="100" alt = 'Not available'/>
                    </NavLink>
                  </div>
                )
            })
          }
        </div> */}

        <Footer />

        <ChatTool />

      </div>

    );
  }
}

const mapStateToProps = (state) => {
  return {
    lang: state.language.lang,
    show_landing_page: state.general.show_landing_page
  }
}

export default withStyles(styles)(connect(mapStateToProps, { authCheckState, handle_referid, hide_landing_page })(Home));