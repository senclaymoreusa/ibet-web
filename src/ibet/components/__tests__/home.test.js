// Home components test

import React from 'react';
import { shallow } from 'enzyme';

import { Home } from '../home';

const setUp = (props={}) => {
    const authCheckState = jest.fn();
    const component = shallow(<Home authCheckState={authCheckState} />);
    return component;
};

const findByTestAtrr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
};

describe('Home Component', () => {

    let component;
    beforeEach(() => {
        component = setUp();
    });

    //Snapshot Test
    it('should render without throwing an error', () => {
        expect(component).toMatchSnapshot();
    });

    //Components Test
    it('should have one header', () => {
        const wrapper = findByTestAtrr(component, 'headerLine');
        expect(wrapper.length).toBe(1);
    });


    it('should render a H1', () => {
        const h1 = findByTestAtrr(component, 'header');
        expect(h1.length).toBe(1);
    });
      
});
 