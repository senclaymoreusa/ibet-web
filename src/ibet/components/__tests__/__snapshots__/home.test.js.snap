// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render without throwing an error 1`] = `
<div>
  <div
    className="rows"
    data-test="headerLine"
  >
    <withRouter(Connect(Navigation)) />
    <div>
      <h1
        data-test="header"
      >
         
        <FormattedMessage
          defaultMessage="Claymore"
          id="home.title"
          values={Object {}}
        />
      </h1>
    </div>
  </div>
  <div
    className="rows"
  />
  <div
    className="rows"
  />
  <div
    className="rows"
  />
</div>
`;
